### PR TITLE
2020 10 event descriptions

### PR DESCRIPTION
**Pull request by nikomatsakis**
New version of https://github.com/alumxi22/website/pull/11 but using a branch

### CODE REVIEW

**Comment by nikomatsakis** [comment]
Ok, @ggilmore, I see -- the command is `dhall format`... I'll dig a bit more into the bash script. :)

**Comment by nikomatsakis** [comment]
OK, the problem seems to be the `parallel` command, actually. I am not sure where this command is coming from. I tried `sudo apt install parallel` and I got some kind of program, but it doesn't accept a `--citation` option, for example, so it's probably not the one you were expecting.

**Comment by nikomatsakis** [comment]
OK I've been digging deeper now. I think I *did* have the right `parallel` command -- and this "citation" thing is bonkers! citation was failing because latex is not installed, and so it can't run bibtex, or something nuts like that. I changed to use `--will-cite` (since we were, after all, just dumping the citation into /dev/null) and now things are *kind* of working.

I then modified the script to not try and format directories, since that was causing `dhall format` to fail.

But *now* I'm getting an error because some command `fd` is not present. I tried installing `sudo apt install fdclone` but that launched some interactive program that was displaying me menus in what appeared to be Japanese characters, so presumably that is not correct. =)

**Comment by nikomatsakis** [comment]
Oh, right, `fd` was this alternative `find`. Maybe I should just give up and install it.

**Comment by nikomatsakis** [comment]
OK, I think I got everything working. Take a look @ggilmore. I did the following things:

* I modified the scripts to use `find` instead of `fd`.
* I updated the instructions to mention the need to install `parallel`.
* I use `--will-cite` to parallel.

Now I can run `just build` and things work. :tada:

**Comment by nikomatsakis** [comment]
The shellcheck errors look pre-existing to me, but I could go ahead and fix them..?

**Comment by nikomatsakis** [comment]
OK, I've fixed the shell warnings, CI is passing, I'm going to merge!

**Comment by ggilmore** [comment]
In all honesty, I would prefer keeping fd. I write several tools using it, and keeping that consistency makes things a lot easier to maintain. In addition, `fd` has better defaults than find (example: respects gitignore patterns) that have bitten me a lot in the past. While `find` might be fine for this commit, I can easily see myself writing more helper scripts that would benefit a lot from using fd instead of find. (example: I'll need to setup a light typescript project inside this repo for the donation page's progress bar - that will require helper scripts for linting, building, formatting etc.). 

What do you think about adding `fd` as another `asdf` requirement to the project readme? That's how it's currently installed for CI.

**Comment by nikomatsakis** [comment]
@ggilmore that works for me, I don't really know much about asdf though.

**Comment by ggilmore** [comment]
The asdf instructions are part of the base readme - are those instructions lacking? Of course, you can install all these tools out of band, but you run the risk of your dev environment diverging from CI's.

**Comment by nikomatsakis** [comment]
@ggilmore I installed the tools as described in the readme. What I meant is, I don't know what updates are needed to the repo to install `fd` through asdf -- are you just saying I should go find `asdf-df` or whatever and run that locally?

**Comment by nikomatsakis** [comment]
Also, is GNU parallel something we can manage through asdf? It'd be nice to have a setup that didn't require people to install that either (or at least we should mention it in the README).

**Comment by ggilmore** [comment]
The repo / CI is already setup to manage `fd` through asdf: 


- https://github.com/alumxi22/website/blob/e0a4aa55e83f173f940deb51675b7cbfe48aa7c4/.tool-versions#L7
- https://github.com/alumxi22/website/blob/e0a4aa55e83f173f940deb51675b7cbfe48aa7c4/scripts/ci/dhall/setup.dhall#L8
- https://github.com/alumxi22/website/blob/e0a4aa55e83f173f940deb51675b7cbfe48aa7c4/scripts/ci/asdf-add-plugins.sh#L17
- https://github.com/alumxi22/website/runs/1277899339#step:4:82

I just don't call out the installation of the `asdf-fd` plugin in the README yet for the local setup instructions. I'd just add another bullet point to install this plugin https://gitlab.com/wt0f/asdf-fd (or since it's been added to https://github.com/asdf-vm/asdf-plugins, `asdf plugin-add fd` will just work - this is what we do in CI).

**Comment by ggilmore** [comment]
> Also, is GNU parallel something we can manage through asdf? It'd be nice to have a setup that didn't require people to install that either (or at least we should mention it in the README).

No, nobody has written a plugin for it yet (nor do I know where the source lives or if static binaries exist for it). I

I see now that this repo has a slightly order version of these scripts. In later changes in other repos, I actually moved the `parallel` calls to a separate script that has a more helpful error message: https://github.com/ggilmore/kickstart/blob/4d60da80f7f3a64ed6f4bbea3ea829771473271e/scripts/parallel_run.sh#L1-L12. I can add the refactor to this repository too.

**Comment by ggilmore** [comment]
`just`, however, is something that can be managed through asdf now that my PR has been accepted https://github.com/asdf-vm/asdf-plugins/pull/287

File: README.md [hunk 1]
```diff
@@ -7,6 +7,7 @@ This project relies on [Hugo](https://gohugo.io/), [Yarn](https://classic.yarnpk
 ## Basic Dev Instructions
 
 1. Run `git submodule update --init --recursive` if you haven't already done so
+1. Install GNU parallel (`sudo apt install parallel` on debian and friends)
```
Comment (cjosephson) [inline README.md:10]: Perhaps we get rid of this for now, or at least have it be separate.

**Review by cjosephson — COMMENTED** [review]
This all looks good to me, minus the README change. I do agree we need to fix the trouble you ran into, but it sounds like perhaps the way forward (e.g. parallel, find vs fd) is still under discussion so the README change should be punted to a later PR.

**Comment by nikomatsakis** [comment]
OK, I removed the use of `find` and the changes related to `parallel`. I left the bash tweaks since CI was testing them and complaining about them. (I still have the changes related to `parallel` locally, I used them to run just build, since I'm not able to do so without them for some reason.)

Regarding parallel, we could presumably just replace `parallel` with for loops, too.

**Comment by cjosephson** [comment]
> OK, I removed the use of `find` and the changes related to `parallel`. I left the bash tweaks since CI was testing them and complaining about them. (I still have the changes related to `parallel` locally, I used them to run just build, since I'm not able to do so without them for some reason.)
> 
> Regarding parallel, we could presumably just replace `parallel` with for loops, too.

ahhh, so close! looks like you forgot to re-render before pushing? that check is failing.

**Comment by ggilmore** [comment]
You should be able to run the command `just` (or `just build`) to render the site and format everything.

**Comment by nikomatsakis** [comment]
OK, I regenerated the site. (Sigh, I had to cherry-pick the two commits to modify gnu parallel and then remove them =)

**Review by cjosephson — APPROVED** [review]

**Pull request merged**